lib/misc/api/api_search.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+import 'api_url.dart' as api;
+import 'dart:developer';
+import 'package:flutter/foundation.dart';
+
+Future<List<dynamic>> search(query) async {
+  List<dynamic> products = [];
+
+  if(query == ""){
+    return Future.error("Empty query");
+  }
+
+  try {
+    await http.get(
+      Uri.parse('${api.getApiBaseUrl()}/products/search/$query'),
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+        'Accept': 'application/json'
+      },
+    ).then((response) {
+      if (response.statusCode == 200) {
+        // If OK response decode response and set to categories variable
+        var json = jsonDecode(response.body)['data'];
+
+        if (kDebugMode) {
+           debugPrint(jsonEncode(jsonDecode(response.body)['data']));
+        }
```

Comment:
The json variable could be made final instead of var, since it isnt changed again.
Also you could reuse the json variable in the debugprint.